Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -152,13 +152,14 @@ NetworkConnectionToWebProcess::NetworkConnectionToWebProcess(NetworkProcess& net
     // Use this flag to force synchronous messages to be treated as asynchronous messages in the WebProcess.
     // Otherwise, the WebProcess would process incoming synchronous IPC while waiting for a synchronous IPC
     // reply from the Network process, which would be unsafe.
-    m_connection->setOnlySendMessagesAsDispatchWhenWaitingForSyncReplyWhenProcessingSuchAMessage(true);
-    m_connection->open(*this);
+    Ref protectedConnection = m_connection;
+    protectedConnection->setOnlySendMessagesAsDispatchWhenWaitingForSyncReplyWhenProcessingSuchAMessage(true);
+    protectedConnection->open(*this);
 #if USE(RUNNINGBOARD)
-    m_connection->setOutgoingMessageQueueIsGrowingLargeCallback([weakThis = WeakPtr { *this }] {
+    protectedConnection->setOutgoingMessageQueueIsGrowingLargeCallback([weakThis = WeakPtr { *this }] {
         ensureOnMainRunLoop([weakThis] {
             if (weakThis)
```

Comment:
Should we capture weakThis with RefPtr?

Reply:
I don't think so. We don't want to keep the whole object alive in the case object gets destructed between the time this function is called and this code has a chance to run in the main runloop.